pkg/api/v1/types.go

```diff
@@ -1312,6 +1312,15 @@ type NodeSpec struct {
 	Unschedulable bool `json:"unschedulable,omitempty" description:"disable pod scheduling on the node; see http://releases.k8s.io/HEAD/docs/admin/node.md#manual-node-administration"`
 }
 
+type KubeletPorts struct {
+	// Port on which Kubelet is listening
+	KubeletPort uint `json:"kubeletPort" description:"port on which Kubelet is listening"`
+	// Read only port open on Kubelet
+	KubeletReadOnlyPorty uint `json:"kubeletReadOnlyPort" description:"read-only port exposed by Kubelet"`
```

Comment:
Also, what is the read-only port used for?

I might mark it `omitempty` to indicate that it's optional, in case we decide to remove it later, and assuming it's possible to gracefully degrade in its absence.